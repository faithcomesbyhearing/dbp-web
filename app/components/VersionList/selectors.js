import { createSelector } from 'reselect';

const getHomepageState = (state) => state['homepage'];

const selectActiveBookId = () =>
	createSelector(getHomepageState, (homepage) => homepage['activeBookId']);

const selectActiveChapter = () =>
	createSelector(getHomepageState, (homepage) => homepage['activeChapter']);

export { selectActiveBookId, selectActiveChapter };
