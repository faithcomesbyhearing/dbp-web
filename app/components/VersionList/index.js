/**
 *
 * VersionList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import Router from 'next/router';
import { createStructuredSelector } from 'reselect';
import LoadingSpinner from '../LoadingSpinner';
import VersionListSection from '../VersionListSection';
import messages from './messages';
import { selectActiveBookId, selectActiveChapter } from './selectors';
import { changeVersion } from '../../containers/HomePage/actions';
import getBookMetaData from '../../utils/getBookMetaData';
import getFirstChapterReference from '../../utils/getFirstChapterReference';
import { selectHasVideo } from '../../containers/VideoPlayer/selectors';
import getUrl from '../../utils/hrefLinkOrAsLink';

export class VersionList extends React.PureComponent {
	get filteredVersionList() {
		const { bibles, activeTextId, filterText, activeChapter, activeBookId } =
			this.props;
		const filteredBibles = filterText
			? bibles?.filter(this.filterFunction)
			: bibles;

		const getKey = (bible) => {
			if (!bible) return '';
			const abbr = bible.get('abbr');
			const date = bible.get('date') || '2019';
			const typeSuffix = bible.get('hasVideo') ? '_video' : '_plain';
			return abbr + date + typeSuffix;
		};

		// Change the way I figure out if a resource has text or audio
		// path, key, types, className, text, clickHandler
		// Set the path to just the bible_id and let app.js handle getting the actual book and chapter needed
		const scrubbedBibles = filteredBibles?.reduce((acc, bible) => {
			const item = {
				path: {
					textId: bible.get('abbr'),
					bookId: activeBookId,
					chapter: activeChapter,
				},
				key: getKey(bible),
				clickHandler: (audioType) =>
					this.handleVersionListClick(bible, audioType),
				className: bible.get('abbr') === activeTextId ? 'active-version' : '',
				title: bible.get('name'),
				text: bible.get('vname') || bible.get('name') || bible.get('abbr'),
				altText:
					bible.get('vname') && bible.get('vname') !== bible.get('name')
						? bible.get('name')
						: '',
				types: bible.get('filesets')
					? bible
							.get('filesets')
							.reduce((a, c) => ({ ...a, [c.get('type')]: true }), {})
					: {},
			};

			acc.push(item);
			return acc;
		}, []);

		// When I first get the response from the server with filesets
		const video = [];
		const audioAndText = [];
		const audioOnly = [];
		const textOnly = [];

		scrubbedBibles?.forEach((b) => {
			if (b.types.video_stream) {
				video.push(b);
			} else if (
				(b.types.audio_drama || b.types.audio) &&
				(b.types.text_plain || b.types.text_format)
			) {
				audioAndText.push(b);
			} else if (b.types.audio_drama || b.types.audio) {
				audioOnly.push(b);
			} else {
				textOnly.push(b);
			}
		});

		const videoComponent = video?.length ? (
			<div className={'version-list-section'} key={'video'}>
				<div className={'version-list-section-title'}>
					<FormattedMessage {...messages.video} />
				</div>
				<VersionListSection items={video} />
			</div>
		) : null;
		const audioAndTextComponent = audioAndText.length ? (
			<div className={'version-list-section'} key={'audio-and-text'}>
				<div className={'version-list-section-title'}>
					<FormattedMessage {...messages.audioAndText} />
				</div>
				<VersionListSection items={audioAndText} />
			</div>
		) : null;
		const audioOnlyComponent = audioOnly.length ? (
			<div className={'version-list-section'} key={'audio-only'}>
				<div className={'version-list-section-title'}>
					<FormattedMessage {...messages.audioOnly} />
				</div>
				<VersionListSection items={audioOnly} />
			</div>
		) : null;
		const textOnlyComponent = textOnly.length ? (
			<div className={'version-list-section'} key={'text-only'}>
				<div className={'version-list-section-title'}>
					<FormattedMessage {...messages.textOnly} />
				</div>
				<VersionListSection items={textOnly} />
			</div>
		) : null;

		const components = [
			videoComponent,
			audioAndTextComponent,
			audioOnlyComponent,
			textOnlyComponent,
		];

		if (bibles?.size === 0) {
			return (
				<span className="version-item-button">
					There was an error fetching this resource, an Admin has been notified.
					We apologize for the inconvenience.
				</span>
			);
		}

		return scrubbedBibles?.length ? (
			components
		) : (
			<span className="version-item-button">
				There are no matches for your search.
			</span>
		);
	}

	filterFunction = (bible) => {
		const lowerCaseText = this.props.filterText.toLowerCase();
		const properties = ['vname', 'name', 'abbr', 'date'];

		return properties.some((property) => {
			const propValue = bible.get(property) || '';
			return propValue.toLowerCase().includes(lowerCaseText);
		});
	};

	// Make async
	handleVersionListClick = async (bible, audioType) => {
		// Figure out correct url here based on session variable value
		const { toggleTextSelection, activeTextId, activeBookId, activeChapter } =
			this.props;
		const hasVideo = !!bible.get('hasVideo');

		if (bible.get('jesusFilm')) {
			// If there is a Jesus Film then use alternate navigation logic
			Router.push(`/jesus-film/${bible.get('iso')}`);
			toggleTextSelection();
			// End function early since navigation should have already occurred
			return;
		}
		// If bible id is equal to the active bible id then just return and don't change version
		if (bible.get('abbr').toLowerCase() === activeTextId.toLowerCase()) {
			toggleTextSelection();
			return;
		}

		if (bible.get('abbr').toLowerCase() !== activeTextId.toLowerCase()) {
			this.props.dispatch(changeVersion({ state: true }));
		}

		if (JSON.parse(sessionStorage.getItem('bible_is_maintain_location'))) {
			Router.push(
				getUrl({
					textId: bible.get('abbr'),
					bookId: activeBookId,
					chapter: activeChapter,
					audioType,
					isHref: true,
				}),
				getUrl({
					textId: bible.get('abbr'),
					bookId: activeBookId,
					chapter: activeChapter,
					audioType,
					isHref: false,
				}),
			);
		} else {
			// Find url and push that one
			const [filteredMetadata, allMetadata] = await getBookMetaData({
				idsForBookMetadata: bible
					.get('filesets')
					.map((set) => [set.get('type'), set.get('id')])
					.toJS(),
			});

			const bookChapterUrl = getFirstChapterReference(
				bible.get('filesets').toJS(),
				hasVideo,
				allMetadata,
				filteredMetadata,
				audioType,
			);

			// Need to parse out the bookChapterUrl to create the href version for the server
			// Safe to access 0th element for \w of match since it always returns a string
			const bookId = bookChapterUrl.match(/\w*/)[0]; // Get first word which is bookId
			const chapterId = bookChapterUrl.match(/\/\w*/)[0].slice(1); // get second part of url
			const query =
				bookChapterUrl.match(/\?.*/) &&
				bookChapterUrl.match(/\?.*/)[0].replace('?', '&'); // Get any query params at the end

			Router.push(
				`/app?bibleId=${bible.get(
					'abbr',
				)}&bookId=${bookId}&chapter=${chapterId}${query || ''}`,
				`/bible/${bible.get('abbr')}/${bookChapterUrl}`,
			);
		}

		if (bible) {
			if (audioType) {
				if (
					typeof window !== 'undefined' &&
					(audioType === 'audio' || audioType === 'audio_drama')
				) {
					document.cookie = `bible_is_audio_type=${audioType};path=/bible/${bible.get(
						'abbr',
					)}`;
				}
			}
			toggleTextSelection();
		}
	};

	render() {
		const { active, loadingVersions } = this.props;

		return (
			<div
				style={{ display: active ? 'block' : 'none' }}
				className="text-selection-section"
			>
				<div className="version-name-list">
					{loadingVersions ? <LoadingSpinner /> : this.filteredVersionList}
				</div>
			</div>
		);
	}
}

VersionList.propTypes = {
	bibles: PropTypes.object,
	dispatch: PropTypes.func,
	setActiveText: PropTypes.func,
	toggleTextSelection: PropTypes.func,
	activeTextId: PropTypes.string,
	filterText: PropTypes.string,
	activeBookId: PropTypes.string,
	activeChapter: PropTypes.number,
	active: PropTypes.bool,
	hasVideo: PropTypes.bool,
	loadingVersions: PropTypes.bool,
};

function mapDispatchToProps(dispatch) {
	return {
		dispatch,
	};
}

const mapStateToProps = createStructuredSelector({
	activeBookId: selectActiveBookId(),
	activeChapter: selectActiveChapter(),
	hasVideo: selectHasVideo(),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withConnect)(VersionList);
