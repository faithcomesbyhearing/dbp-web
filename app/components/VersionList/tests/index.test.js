import React from 'react';
import { fromJS } from 'immutable';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'; // for additional matchers like toHaveAttribute
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import VersionList from '..';
import { getTexts } from '../../VersionListSection/tests/versionListSectionUtils';

// Mock the selectors
jest.mock('../selectors', () => ({
	selectActiveBookId: jest.fn(() => () => 'MAT'), // Return a function
	selectActiveChapter: jest.fn(() => () => 5), // Return a function
	selectHasVideo: jest.fn(() => () => false), // Return a function
}));

jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>, // eslint-disable-line react/prop-types
	defineMessages: (messages) => messages,
}));

// Create a mock Redux store
const mockStore = configureMockStore();

const props = {
	dispatch: jest.fn(),
	setActiveText: jest.fn(),
	toggleTextSelection: jest.fn(),
	activeTextId: 'ENGESV',
	activeBookId: 'MAT',
	activeChapter: 5,
	filterText: '',
	active: true,
	loadingVersions: false,
};

let bibles = fromJS([]);

describe('<VersionList />', () => {
	beforeEach(async () => {
		const jsBibles = await getTexts({ languageCode: 6414 });
		bibles = fromJS(jsBibles);
	});

	it('Should match previous snapshot', () => {
		const initialState = fromJS({
			homepage: {
				hasVideo: false, // This is where your selector looks for `hasVideo`
			},
			videoPlayer: {
				videoList: [],
			},
		});

		const store = mockStore(initialState);

		const { asFragment } = render(
			<Provider store={store}>
				<VersionList {...props} bibles={bibles} />
			</Provider>,
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('Should render a list of version names', () => {
		const initialState = fromJS({
			homepage: {
				hasVideo: false, // This is where your selector looks for `hasVideo`
			},
			videoPlayer: {
				videoList: [],
			},
		});

		const store = mockStore(initialState);

		render(
			<Provider store={store}>
				<VersionList {...props} bibles={bibles} />
			</Provider>,
		);

		// Add assertions here if needed.
	});

	it('Should render all available bibles as accordion titles', () => {
		const initialState = fromJS({
			homepage: {
				hasVideo: false, // This is where your selector looks for `hasVideo`
			},
			videoPlayer: {
				videoList: [],
			},
		});

		const store = mockStore(initialState);

		const { container } = render(
			<Provider store={store}>
				<VersionList {...props} bibles={bibles} />
			</Provider>,
		);

		const versionList = container.querySelector('.version-name-list');

		expect(versionList).toBeInTheDocument();
		// Add assertions here if needed.
	});
});
