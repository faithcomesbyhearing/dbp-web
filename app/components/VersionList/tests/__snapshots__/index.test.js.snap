// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VersionList /> Should match previous snapshot 1`] = `
<div
  className="text-selection-section"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="version-name-list"
  >
    <div
      className="version-list-section"
    >
      <div
        className="version-list-section-title"
      >
        <span>
          Video
        </span>
      </div>
      <div
        className="accordion"
        role="tablist"
      >
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-0"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-0"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <h4
              className=""
              title="New International Version"
            >
              New International Version
            </h4>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-0"
            className="accordion-body-style accordion__body--hidden"
            id="accordion__body-0"
            role="tabpanel"
          >
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Dramatized Version
            </a>
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Non-Dramatized Version
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="version-list-section"
    >
      <div
        className="version-list-section-title"
      >
        <span>
          Audio & Text
        </span>
      </div>
      <div
        className="accordion"
        role="tablist"
      >
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-1"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-1"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <h4
              className="active-version"
              title="English Standard Version"
            >
              English Standard Version
            </h4>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-1"
            className="accordion-body-style accordion__body--hidden"
            id="accordion__body-1"
            role="tabpanel"
          >
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Dramatized Version
            </a>
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Non-Dramatized Version
            </a>
          </div>
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-2"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-2"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <h4
              className=""
              title="King James Version"
            >
              King James Version
            </h4>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-2"
            className="accordion-body-style accordion__body--hidden"
            id="accordion__body-2"
            role="tabpanel"
          >
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Dramatized Version
            </a>
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Non-Dramatized Version
            </a>
          </div>
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-3"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-3"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="New American Bible"
            >
              New American Bible
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-3"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-3"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-4"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-4"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="World English Bible (Hosanna audio)"
            >
              World English Bible (Hosanna audio)
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-4"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-4"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-5"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-5"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="World English Bible (Afred Henson)"
            >
              World English Bible (Afred Henson)
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-5"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-5"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-6"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-6"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <h4
              className=""
              title="New International Version (Anglicised)"
            >
              New International Version (Anglicised)
            </h4>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-6"
            className="accordion-body-style accordion__body--hidden"
            id="accordion__body-6"
            role="tabpanel"
          >
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Dramatized Version
            </a>
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Non-Dramatized Version
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="version-list-section"
    >
      <div
        className="version-list-section-title"
      >
        <span>
          Audio Only
        </span>
      </div>
      <div
        className="accordion"
        role="tablist"
      >
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-7"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-7"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="Contemporary English Version"
            >
              Contemporary English Version
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-7"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-7"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-8"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-8"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <h4
              className=""
              title="New Revised Standard Version"
            >
              New Revised Standard Version
            </h4>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-8"
            className="accordion-body-style accordion__body--hidden"
            id="accordion__body-8"
            role="tabpanel"
          >
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Dramatized Version
            </a>
            <a
              className="version-item-button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              Non-Dramatized Version
            </a>
          </div>
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-9"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-9"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="Contemporary English Version (African American Voices)"
            >
              Contemporary English Version (African American Voices)
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-9"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-9"
            role="tabpanel"
          />
        </div>
      </div>
    </div>
    <div
      className="version-list-section"
    >
      <div
        className="version-list-section-title"
      >
        <span>
          Text Only
        </span>
      </div>
      <div
        className="accordion"
        role="tablist"
      >
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-10"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-10"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="New American Standard Bible"
            >
              New American Standard Bible
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-10"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-10"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-11"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-11"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="New Life Version"
            >
              New Life Version
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-11"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-11"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-12"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-12"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="English Version for the Deaf"
            >
              English Version for the Deaf
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-12"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-12"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-13"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-13"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="American Standard Version"
            >
              American Standard Version
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-13"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-13"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-14"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-14"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="World Messianic Version"
            >
              World Messianic Version
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-14"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-14"
            role="tabpanel"
          />
        </div>
        <div
          className="accordion-title-style"
        >
          <div
            aria-controls="accordion__body-15"
            aria-selected={false}
            className="accordion__title"
            disabled={false}
            id="accordion__title-15"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="tab"
            tabIndex="0"
          >
            <a
              className=" top-level-title"
              onClick={[Function]}
              role="button"
              tabIndex={0}
              title="Revised Version 1885"
            >
              Revised Version 1885
            </a>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="accordion__title-15"
            className="accordion__body accordion__body--hidden"
            id="accordion__body-15"
            role="tabpanel"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
